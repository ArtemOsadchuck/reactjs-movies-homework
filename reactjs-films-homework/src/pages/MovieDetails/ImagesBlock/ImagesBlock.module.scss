.imagesBlockWrapper {
  color: rgb(247, 247, 247);
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  margin-top: 2rem;
  width: 68%;

  .imagesTitleWrapper {
    align-items: center;
    display: flex;
    margin-bottom: 1rem;
    width: 48rem;

    @media screen and (max-width: 1220px) {
      padding: 0 1.5rem;
    }
  }

  @media screen and (max-width: 1220px) {
    justify-content: center;
    margin: 2rem auto;
    padding: 0 1.5rem;
    width: 90%;
  }

  @media screen and (max-width: 1020px) {
    padding: 0;
    width: 100%;
  }
}

.imagesWrapper {
  column-gap: 0.5rem;
  display: flex;
  flex-wrap: wrap;
  grid-gap: 0.75rem;
  width: 100%;

  @media screen and (max-width: 1220px) {
    justify-content: center;
    margin: 0 auto;
    padding: 0 1.5rem;
  }

  .image {
    border-radius: 5px;
  }
}
