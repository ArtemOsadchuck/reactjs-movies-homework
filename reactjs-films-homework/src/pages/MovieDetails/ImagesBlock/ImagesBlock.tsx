import React from 'react';
import styles from './ImagesBlock.module.scss';

export interface IImagesBlock {
  images: IImagesBlockProps[];
  title: string;
  imgWidth: string;
  imagesQuality: number;
}
export interface IImagesBlockProps {
  aspect_ratio?: number;
  file_path: string;
  height?: number;
  iso_639_1?: null;
  vote_average?: number;
  vote_count?: number;
  width?: number;
}

const ImagesBlock: React.FC<IImagesBlock> = ({
  images,
  title,
  imagesQuality,
  imgWidth,
}) => {
  return (
    <div className={styles.imagesBlockWrapper}>
      <div className={styles.imagesTitleWrapper}>
        <h3 className={styles.imagesTitle}>{title}</h3>
      </div>
      <div className={styles.imagesWrapper}>
        {images.length
          ? images.slice(0, imagesQuality).map((el) => {
              return (
                <img
                  className={styles.image}
                  width={imgWidth}
                  key={el.file_path.length * Math.random()}
                  src={`https://image.tmdb.org/t/p/w500/${el.file_path}`}
                  alt="images"
                />
              );
            })
          : null}
      </div>
    </div>
  );
};

export default ImagesBlock;
