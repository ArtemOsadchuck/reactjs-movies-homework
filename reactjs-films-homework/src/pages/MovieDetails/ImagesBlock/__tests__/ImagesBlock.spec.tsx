import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';

import ImagesBlock, { IImagesBlock } from '../ImagesBlock';
import ImagesBlockMocks from '../mocks';

import RouterWrapper from '../../../../__testsUtils__/routerHoc';
import StoreWrapper from '../../../../__testsUtils__/storeHoc';

const renderComponent = ({
  title,
  images,
  imagesQuality,
  imgWidth,
}: IImagesBlock) => {
  const urlApp = '/movie-details';

  const { asFragment } = render(
    <StoreWrapper>
      <RouterWrapper url={urlApp}>
        <ImagesBlock
          images={images}
          title={title}
          imgWidth={imgWidth}
          imagesQuality={imagesQuality}
        />
      </RouterWrapper>
    </StoreWrapper>
  );
  return asFragment();
};

describe('ImagesBlock', () => {
  it('should renders correctly', () => {
    const imagesBlockData = {
      title: 'images',
      imagesQuality: 8,
      imgWidth: '172px',
      images: ImagesBlockMocks,
    };
    const fragment = renderComponent(imagesBlockData);
    const imgLength = screen.getAllByRole('img').length;

    expect(fragment).toMatchSnapshot();
    expect(imgLength).toEqual(2);
  });

  cleanup();
});
