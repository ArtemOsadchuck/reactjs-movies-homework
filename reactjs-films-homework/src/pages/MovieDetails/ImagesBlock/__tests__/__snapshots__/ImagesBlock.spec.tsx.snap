// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagesBlock should renders correctly 1`] = `
<DocumentFragment>
  <div
    class="imagesBlockWrapper"
  >
    <div
      class="imagesTitleWrapper"
    >
      <h3
        class="imagesTitle"
      >
        images
      </h3>
    </div>
    <div
      class="imagesWrapper"
    >
      <img
        alt="images"
        class="image"
        src="https://image.tmdb.org/t/p/w500//irlfhYtHfhZuYpsq2LAoh308NFe.jpg"
        width="172px"
      />
      <img
        alt="images"
        class="image"
        src="https://image.tmdb.org/t/p/w500//iNh3BivHyg5sQRPP1KOkzguEX0H.jpg"
        width="172px"
      />
    </div>
  </div>
</DocumentFragment>
`;
